user_management/ui/views.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib import messages
+from django.contrib.auth import get_user_model
+from django.contrib.auth.tokens import default_token_generator
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+from django.utils.encoding import force_text
+from django.utils.http import urlsafe_base64_decode
+from django.views import generic
+
+
+class VerifyUserEmailView(generic.RedirectView):
+    """
+    A view which verifies a user's email address.
+
+    Accessed via a link in an email sent to the user, which contains both a
+    uid generated from the user's pk, and a token also generated from the user
+    object, for verification.  If everything lines up, it makes the user
+    active.
```

Comment:
It's not quite that strict, but it might as well be.  It relies on the user's `pk` and `last_login`.  The `pk` uniquely identifies the user object, and since you can't change stuff about a user without them being logged in*, I think that gives us effective coverage against potential changes.  Certainly it's good enough for Django itself, it'd seem.

*Or without having admin access, but that's not something we can protect against or care about here, since the admin can sidestep this entire process easily.